### PR TITLE
Wait for async processes to configure `buildScan` extension

### PR DESCRIPTION
**Pull request by bamboo**
By running the git commands in parallel but waiting on each `Future`
value sequentially.
 
- [ ] Verify test coverage and [CI build status](https://builds.gradle.org/viewQueued.html?itemId=12546282)

### CODE REVIEW

File: buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt [hunk 1]
```diff
@@ -102,46 +103,58 @@ open class BuildScanPlugin : Plugin<Project> {
...
-        }
+                .run {
+                    assert(waitFor() == 0)
+                    inputStream.bufferedReader().use { it.readText().trim() }
+                }
 
-        fun execAsync(action: () -> Unit) {
-            val latch = CountDownLatch(1)
-            thread(start = true) {
+        val threadPool = Executors.newCachedThreadPool()
```
Comment (big-guy) [inline buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt]: do we need to explicitly shutdown the threadpool now so we don't leak thread pools across builds?

**Comment by bamboo** [comment]
Thanks, @big-guy.

It turned out the `buildScan` extension is thread-safe so I changed the code back to use threads and added the required `Thread::join` calls.

**Comment by ldaley** [comment]
@bamboo did you verify this isn’t slower?

**Comment by bamboo** [comment]
The performance test failed so I reverted it.

The PR adds two things that might account for the longer build time:
 - one more thread
 - the `Thread.join` calls

I've reduced the parallelism back to the original level and started a new build. Let's see.

**Comment by ldaley** [comment]
With the previous version, Gradle could go on to other scripts while the processes are running. With yours, we block and wait before moving on to other scripts.

**Comment by ldaley** [comment]
That’s why the original intent was to wait on a latch in buildScan.buildFinished as that guarantees everything has finished before transmitting the event stream.

**Comment by ldaley** [comment]
You could achieve the same by doing the thread join in place of the multiple latch wait.

**Comment by bamboo** [comment]
Ah, `buildScan.buildFinished`, nice! 👍

**Comment by bamboo** [comment]
Superseded by #5224.

**Pull request merged**